internal/collector/cloudwatch.go

```diff
@@ -1,57 +1,93 @@
 package collector
 
 import (
+	_ "embed"
+	"html/template"
+	"strings"
+
 	log "github.com/ViaQ/logerr/v2/log/static"
 	obs "github.com/openshift/cluster-logging-operator/api/observability/v1"
 	"github.com/openshift/cluster-logging-operator/internal/api/observability"
 	"github.com/openshift/cluster-logging-operator/internal/collector/common"
 	"github.com/openshift/cluster-logging-operator/internal/constants"
 	"github.com/openshift/cluster-logging-operator/internal/generator/vector/output/cloudwatch"
-	v1 "k8s.io/api/core/v1"
+	"github.com/openshift/cluster-logging-operator/internal/reconcile"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	"github.com/openshift/cluster-logging-operator/internal/utils"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	CloudwatchCredentialsTemplate = template.Must(template.New("cw credentials").Parse(cloudwatchCredentialsTemplateStr))
+	//go:embed cloudwatch.credentials.tmpl
+	cloudwatchCredentialsTemplateStr string
 )
 
-// Add volumes and env vars if output type is cloudwatch and role is found in the secret
-func addWebIdentityForCloudwatch(collector *v1.Container, forwarderSpec obs.ClusterLogForwarderSpec, secrets observability.Secrets) {
-	if secrets == nil {
-		return
+type CloudwatchWebIdentity struct {
+	Name    string
+	RoleARN string
+	Token   string
+}
+
+// ReconcileAWSCredentialsConfigMap reconciles a configmap with credential profile(s) for Cloudwatch output(s).
+func ReconcileAWSCredentialsConfigMap(k8sClient client.Client, reader client.Reader, namespace, name string, outputs []obs.OutputSpec, secrets observability.Secrets, configMaps map[string]*corev1.ConfigMap, owner metav1.OwnerReference) (*corev1.ConfigMap, error) {
```

Comment:
Do we need the uncached reader other then k8sClient here?  Usually that is only an issue when trying to read a cluster resource that we don't own

Reply:
Well the `reader` is used in [reconcile.configmap](https://github.com/openshift/cluster-logging-operator/blob/abf40334e35c27579a6ca98af5ec16749969cda2/internal/reconcile/configmaps.go#L20) probably because it is used for reconciling the `dashboard` configmap.

Reply:
The dashboard, however, lives in an unmanaged namespace.  You should not need to pass the reader here and simply pass the client as the reader in the dependent function

Reply:
I see what you mean. Will make the change